Cython/Compiler/PyrexTypes.py

```diff
@@ -192,12 +192,16 @@ class PyrexType(BaseType):
     #  is_string             boolean     Is a C char * type
     #  is_pyunicode_ptr      boolean     Is a C PyUNICODE * type
     #  is_cpp_string         boolean     Is a C++ std::string type
+    #  is_indexed_pytype     boolean     Is a Python type identifier that can be indexed/"templated"
     #  is_unicode_char       boolean     Is either Py_UCS4 or Py_UNICODE
     #  is_returncode         boolean     Is used only to signal exceptions
     #  is_error              boolean     Is the dummy error type
     #  is_buffer             boolean     Is buffer access type
     #  is_pythran_expr       boolean     Is Pythran expr
     #  is_numpy_buffer       boolean     Is Numpy array buffer
+    #  is_dataclasses_initvar        boolean     Is a dataclasses InitVar
+    #  is_typing_classvar            boolean     Is a typing ClassVar
+    #  is_typing_optional            boolean     Is a typing Optional
```

Comment:
I understand where these are coming from, but `is_typing_optional` totally reads like "is typing optional?", not like a reference to the `typing` module. Not sure right now what a better name would be, but this scheme also doesn't seem to scale well.

Reply:
I've replaced it with `is_special_python_type_constructor` which covers all of them, and then just done `type.name=="typing.Optional"`